circe-display-images.el

```diff
@@ -0,0 +1,197 @@
+;;; -*- lexical-binding: t -*-
+
+;;; circe-display-images.el --- Display images in the channel
+
+;; Copyright (C) 2017 Nathan Aclander
+
+;; Author: Nathan Aclander <nathan.aclander@gmail.com>
+
+;; This file is part of Circe.
+
+;; This program is free software; you can redistribute it and/or
+;; modify it under the terms of the GNU General Public License
+;; as published by the Free Software Foundation; either version 3
+;; of the License, or (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program; if not, write to the Free Software
+;; Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+;; 02110-1301  USA
+
+;;; Commentary:
+
+;; This Circe modules adds the ability to display various image types when
+;; they are linked in a channel. Images are inserted on new lines after
+;; the message containing the URLs. This module requires ImageMagcik.
+
+;; To use it, put the following into your .emacs:
+
+;; (require 'circe-display-images)
+;; (enable-circe-display-images)
+
+;;; Code:
+
+(require 'circe)
+(require 'url)
+
+;;;###autoload
+(defun enable-circe-display-images ()
+  "Enable the Display Images module for Circe.
+This module displays various image types when they are linked in a channel"
+  (interactive)
+  (dolist (buf (buffer-list))
+    (with-current-buffer buf
+      (when (eq major-mode 'circe-channel-mode)
+        (add-circe-display-images))))
+  (add-hook 'circe-channel-mode-hook
+            'add-circe-display-images))
+
+(defun disable-circe-display-images ()
+  "Disable the Display Images module for Circe.
+See `enable-circe-display-images'."
+  (interactive)
+  (dolist (buf (buffer-list))
+    (with-current-buffer buf
+      (when (eq major-mode 'circe-channel-mode)
+        (remove-circe-display-images))))
+  (remove-hook 'circe-channel-mode-hook
+               'add-circe-display-images))
+
+(defun add-circe-display-images ()
+  "Add `circe-display-images' to `lui-pre-output-hook'."
+  (add-hook 'lui-pre-output-hook 'circe-display-images))
+
+(defun remove-circe-display-images ()
+  "Remove `circe-display-images' from `lui-pre-output-hook'."
+  (remove-hook 'lui-pre-output-hook 'circe-display-images))
+
+(defgroup circe-display-images nil
+  "Image display properties for Circe"
+  :prefix "circe-display-images"
+  :group 'circe)
+
+(defcustom circe-display-images-image-regex
+  "\\(https?:\/\/.*?\.\\\(?:png\\|jpg\\|jpeg\\|svg\\|gif\\)\\)"
```

Comment:
This will match stuff like "https://www.google.com/ – I can't find foo.png, anyone?". At least restrict the `.*?` to non-whitespace.

Reply:
That seems like a fair compromise. FWIW, "https://www.google.com/ – I can't find foo.png, anyone?" or "http://www.google.com/fakeImage.jgp" will both fail to download and be a no-op.